tests/src/test/scala/scalaz/EphemeralStreamTest.scala

```diff
@@ -132,4 +132,15 @@ object EphemeralStreamTest extends SpecLite {
       EphemeralStream.fromStream(Stream.iterate(0)(_ + 1).zipWithIndex.take(n))
     )
   }
+
+  "reading from a stream in parallel should be safe" in {
+    import scalaz.concurrent._
+    val limit = 10000000
+    val stm = EphemeralStream.range(1, limit + 1)
```

Comment:
Due to #1259, will need `.memoized` at the end of this line.  Else, `weakMemo` will not be used here.